informer/src/informer_service/informer_service_handler.rs

```diff
@@ -102,3 +116,46 @@ impl InformerService {
         }
     }
 }
+
+pub struct BlockCreatedEventGenerator {
+    client: Arc<Client>,
+    rate_limiter: Limiter,
+}
+
+impl BlockCreatedEventGenerator {
+    pub fn new(client: Arc<Client>) -> Self {
+        Self {
+            client,
+            rate_limiter: ratelimit::Builder::new().quantum(1).interval(Duration::new(0, 5000)).build(),
+        }
+    }
+
+    pub async fn async_runtime_wait(&mut self) {
+        self.rate_limiter.wait();
+    }
+
+    pub fn run(mut self, connection: Connection, mut favorite_block_number: u64) -> tokio::task::JoinHandle<()> {
+        task::spawn(async move {
+            loop {
+                let best_block_number = self.client.best_block_header().number();
+                if best_block_number >= favorite_block_number {
+                    let event = self.gen(favorite_block_number);
+                    connection.cold_notification(&event);
+                    self.async_runtime_wait().await;
+                    favorite_block_number += 1;
```

Comment:
Please call sleep when the best_bock_number < favorite_block_number.

```suggestion
                let best_block_number = self.client.best_block_header().number();
                if best_block_number >= favorite_block_number {
                    let event = self.gen(favorite_block_number);
                    connection.cold_notification(&event);
                    self.async_runtime_wait().await;
                    favorite_block_number += 1;
                } else {
                    sleep();
                }   
```